src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -41,6 +41,56 @@ namespace Utils
             m_db.reset(dbRawPtr);
         }
 
+        /**
+         * @brief Copy constructor.
+         *
+         * @param other Other instance.
+         */
+        RocksDBWrapper(RocksDBWrapper& other)
+            : m_db {std::move(other.m_db)}
```

Comment:
I think this constructor leaves the state of the smart pointer in the `other` instance, valid but undefined by the `std::move` operation.

Taking a smart pointer by reference should be done in order to reset the smart pointer according to core guidelines [R.33](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#r33-take-a-unique_ptrwidget-parameter-to-express-that-a-function-reseats-the-widget).

Reply:
As far as I could read, the `move` operation should reset the `unique_ptr`
https://stackoverflow.com/questions/36071220/what-happens-to-unique-ptr-after-stdmove

But still, what I really want is to be able to move an instance of the RocksDB Wrapper.
The current implementation doesn't allow it because there is an `unique_ptr` as a member variable.
 Could be there another alternative?